### DISCUSSION TITLE
How to identify a key with multiple webhook

### DISCUSSION THREAD

**Discussion by pauloanalista** [salman0ansari/whatsapp-api-nodejs#375] [Q&A] [answered]
How to identify a key with multiple webhook

I create my key this way
http://sp.com.br:3333/instance/init?token=TOKEN&key=paulo&webhook=true&webhookUrl=http://sp.com.br/z/ret.php

webhook only returns

{
    "type": "connection",
    "body": {
      "connection": "close"
    }
}

{
    "type": "connection",
    "body": {
      "connection": "open"
    }
}

Can someone help me?

**Comment by diemesolutions** [comment]
Estou no mesmo dilema. Identificar a instância no Webhook.

**Reply by pauloanalista** [reply]
Quando criamos desta forma que eu mencionei, é possível obter esses dados via webhook que foi passado por parametro.

Só não estava funcionando pq minha api não estava em https

**Reply by diemesolutions** [reply]
Sim o meu webhook retorna tudo, mas por exemplo: o webhook retorna desse jeito pra mim 
quando recebo msg de texto

    "type": "message",
    "body": {
        "key": {
            "remoteJid": "meutelefone@s.whatsapp.net",
            "fromMe": false,
            "id": "iddamensagem----------------------------"
        },
        "messageTimestamp": 1666808739,
        "pushName": "dieme.solutions",
        "message": {
            "conversation": "OK",
            "messageContextInfo": {
                "deviceListMetadata": {
                    "senderKeyHash": "EiMnTmV0cchoPg==",
                    "senderTimestamp": "1666641211",
                    "recipientKeyHash": "Cnm6ETs63Z\/25w==",
                    "recipientTimestamp": "1666736551"
                },
                "deviceListMetadataVersion": 2
            }
        },
        "verifiedBizName": "dieme.solutions",
        "text": {
            "messages": [
                {
                    "key": {
                        "remoteJid": "meutelefone@s.whatsapp.net",
                        "fromMe": false,
                        "id": "iddamensagem----------------------------"
                    },
                    "messageTimestamp": 1666808739,
                    "pushName": "dieme.solutions",
                    "message": {
                        "conversation": "OK",
                        "messageContextInfo": {
                            "deviceListMetadata": {
                                "senderKeyHash": "EiMnTmV0cchoPg==",
                                "senderTimestamp": "1666641211",
                                "recipientKeyHash": "Cnm6ETs63Z\/25w==",
                                "recipientTimestamp": "1666736551"
                            },
                            "deviceListMetadataVersion": 2
                        }
                    },
                    "verifiedBizName": "dieme.solutions"
                }
            ],
            "type": "notify"
        },
        "msgContent": ""
    }
}

**Reply by diemesolutions** [reply]
Dessa forma eu recebo a msg mas meu webhook não tem como tratar quem recebeu;

**Comment by pauloanalista** [comment]
Vc terá um webhook por usuário conectado.

**Reply by diemesolutions** [reply]
Sim, eu tenho 5 instâncias conectadas. Mas no retorno do json não informa quem é a instância que recebeu, apenas o número, queria ver se é possível alterar pra esse campo vir junto..

https://chat.whatsapp.com/LCNkcD8IOVyCUPhpi7UYxU

**Reply by diemesolutions** [reply]
Eu não queria criar um webhook pra cada conexão. Queria deixar um webhook pra todos;

**Reply by pauloanalista** [reply]
Só você criar um endpoint que você passa o usuário por parametro.

Vai receber todas as requisições no mesmo endpoint

**Reply by diemesolutions** [reply]
Eu fiz minha URL de webhook baseado nesse padrão

http://url do dominio/webhook/{instance_key}

serviu - mas para garantir fiz um cron para de 1 em 1 hra ele fazer o /INIT novamente startando o webhook já que ao reiniciar ele pega o
http://url do dominio/webhook/
que defini lá no .env

observei que o webhook não está captando todos os retornos, não sei se seria do servidor ou algo em particular, mas já dá pra trabalhar em cima disso.

fizemos um grupo para interação dessa API https://chat.whatsapp.com/LCNkcD8IOVyCUPhpi7UYxU
pra ajudar na divulgação e conhecimento mais amplo.

**Reply by pauloanalista** [reply]
Vou entrar no grupo

**Comment by came** [comment]
Maybe the instance id (key) could be added to the body of the SendWebhook function? here: 
https://github.com/salman0ansari/whatsapp-api-nodejs/blob/8c966e5aba683496ac64bc7570d5851ae49d58cc/src/api/class/instance.js#L61

**Comment by salman0ansari** [comment, marked as answer]
fixed in the latest commit https://github.com/salman0ansari/whatsapp-api-nodejs/commit/f90287260cd3395d2ea2234a3f61bdc6e513397f
